inmemory_backend.go

```diff
@@ -25,10 +25,11 @@ func (b *inMemoryBackend) GetResult(taskID string) (*ResultMessage, error) {
 	return nil, errors.New("task does not exist")
 }
 
-func (b *inMemoryBackend) SetResult(taskID string, result *ResultMessage) error {
+func (b *inMemoryBackend) SetResult(taskID string, res *ResultMessage) error {
 	b.lock.Lock()
 	defer b.lock.Unlock()
-	b.ResultStore[taskID] = result
+	result := *res
```

Comment:
This is only a problem in tests since the result message is being cleaned in the message pool (it won't affect any production code as in production the stored result was never used), I had the fix for this in my closed pr, please apply the same fix here which is to copy the message, (forgot that I had that fix there)
https://github.com/vimukthi-git/gocelery/blob/5698763c4514dc36e2600e31bd8e9de36801844c/inmemory_backend.go#L28

Reply:
No need to do anything about it in this pr, but these message pools are a useless optimisation. We can remove them later.